src/index.js

```diff
@@ -1 +1,34 @@
-export default () => true
+import defaultsDeep from 'lodash.defaultsdeep';
+import pickBy from 'lodash.pickby';
+import makeOkCheck from './make-ok-check';
+import buildOptions from './build-options';
+import resolveUrl from './resolve-url';
+
+class Stipulate {
+
+  constructor(options) {
+    this.baseOptions = options;
+  }
+
+  beforeRequest(url, options) {
+    return [url, options];
+  }
+
+  afterResponse(response) {
+    return response;
+  }
+
+  send(urlString, options, query) {
+    const config = buildOptions(options, this.baseOptions);
+    const url = resolveUrl(urlString, query);
+
+    const requestArguments = this.beforeRequest(url, config);
+    const checkOk = makeOkCheck(config);
+
+    return fetch(...requestArguments)
+      .then(checkOk)
+      .then(this.afterResponse);
```

Comment:
Yeah.

I would not call `beforeResponse` and `afterResponse` directly in `send` since `send` is the core of Stipulate and shouldn't be extended/overwritten (If you do extend it, it'd have to be completely rewritten). Instead, I would do `_beforeResponse` and `_afterResponse` that check for the existence of `beforeResponse` and `afterResponse` respectively, that they are functions, and call them if so.

Doing this has two advantages:

1) The hooks are isolated from `send`, and you don't need a default hook that does a passthrough.
2) You wouldn't need to write tests for defaults that don't do anything. Instead, you'd write tests that assert the hooks were called if they were set. This is good because the tests actually match the intended use.